test/unit/finance/charging_test.rb

```diff
@@ -1,89 +1,92 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class Finance::ChargingTest < ActiveSupport::TestCase
+  setup do
+    Invoice.any_instance.stubs log_entry_created: true
+    @provider_account = FactoryBot.create(:provider_account, billing_strategy: FactoryBot.create(:postpaid_billing))
 
-  context 'Invoice#charge! ... ' do
-    setup do
-      Invoice.any_instance.stubs log_entry_created: true
-      @provider_account = FactoryBot.create(:provider_account,
-                                  :billing_strategy => FactoryBot.create(:postpaid_billing))
+    @provider_account.bought_plan.enable_feature!(:postpaid_billing)
+    @plan = FactoryBot.create(:application_plan, issuer: @provider_account.default_service)
 
-      @provider_account.bought_plan.enable_feature!(:postpaid_billing)
-      @plan = FactoryBot.create(:application_plan, :issuer => @provider_account.default_service)
+    @buyer_account = FactoryBot.create(:buyer_account, provider_account: @provider_account)
+    @cinstance = @buyer_account.buy!(@plan)
+  end
 
-      @buyer_account = FactoryBot.create(:buyer_account, :provider_account => @provider_account)
-      @cinstance = @buyer_account.buy!(@plan)
+  class BuyerNotPayingMonthlyTest < Finance::ChargingTest
+    def setup
+      super
```

Comment:
I just checked. When inheriting from a class, the setup callbacks registered with the superclass are executed before the setup method. So no need for super. Because of this is is also alright just to use `setup do` in the subclass.